wsproto/connection.py

```diff
@@ -236,14 +236,22 @@ def receive_bytes(self, data):
     def _process_upgrade(self, data):
         self._upgrade_connection.receive_data(data)
         while True:
-            event = self._upgrade_connection.next_event()
+            try:
+                event = self._upgrade_connection.next_event()
+            except h11.RemoteProtocolError:
+                return ConnectionFailed(CloseReason.PROTOCOL_ERROR,
+                                        "Bad HTTP message"), b''
             if event is h11.NEED_DATA:
                 break
-            elif self.client and isinstance(event, h11.InformationalResponse):
+            elif self.client and isinstance(event, (h11.InformationalResponse,
+                                                    h11.Response)):
```

Comment:
What's the reasoning behind treating `InformationalResponse` and `Response` the same here?